core/src/main/java/org/infinispan/commands/functional/AbstractWriteKeyCommand.java

```diff
@@ -3,35 +3,45 @@
 import static org.infinispan.commons.util.Util.toStr;
 
 import org.infinispan.commands.CommandInvocationId;
+import org.infinispan.commands.InitializableCommand;
+import org.infinispan.commands.functional.functions.InjectableComponent;
 import org.infinispan.commands.write.AbstractDataWriteCommand;
 import org.infinispan.commands.write.ValueMatcher;
 import org.infinispan.encoding.DataConversion;
 import org.infinispan.factories.ComponentRegistry;
 import org.infinispan.functional.impl.Params;
 
-public abstract class AbstractWriteKeyCommand<K, V> extends AbstractDataWriteCommand implements FunctionalCommand<K, V> {
+public abstract class AbstractWriteKeyCommand<K, V> extends AbstractDataWriteCommand implements FunctionalCommand<K, V>, InitializableCommand {
 
    Params params;
    ValueMatcher valueMatcher;
    boolean successful = true;
    DataConversion keyDataConversion;
    DataConversion valueDataConversion;
+   Object function;
```

Comment:
Subclasses have a stronger type than `Object`, can you improve this?

Reply:
Agreed, probably through templating, as it seems there is no common "root" interface (some are Functions and some are InjectableComponent).